pkg/eth/ipld_retriever.go

```diff
@@ -432,6 +432,7 @@ func (r *IPLDRetriever) RetrieveAccountByAddressAndBlockNumber(address common.Ad
 func (r *IPLDRetriever) RetrieveStorageAtByAddressAndStorageKeyAndBlockHash(address common.Address, storageLeafKey, hash common.Hash) (string, []byte, error) {
 	storageResult := new(nodeInfo)
 	stateLeafKey := crypto.Keccak256Hash(address.Bytes())
+	storageLeafKey = crypto.Keccak256Hash(storageLeafKey.Bytes())
```

Comment:
The name of this function implies the actual storage leaf key is provided to it, it shouldn't need to hash it again. If an actual storage key was provided this hashing would break the function. We should perform the hashing of the slot key into the leaf key at the level above or change the name to `RetrieveStorageAtByAddressAndStorageSlotAndBlockHash` so that the name aligns with the behavior.

Reply:
Updated it to `RetrieveStorageAtByAddressAndStorageSlotAndBlockHash` since the api gives us storage slot.